### ISSUE TITLE
Error: Type 'ImageDecoderCallback' not found.

### ISSUE THREAD

**Issue by chinmaygit** [GetStream/stream-chat-flutter#1600]
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

iOS, Android

### What version are you using?

v6.3.0

### What happened?

`stream_photo_gallery_tile` file causing failure at build time because of following code
This is happening on flutter v3.3.10
```
  @override
  ImageStreamCompleter loadImage(
    MediaThumbnailProvider key,
    ImageDecoderCallback decode,
  ) {...
  ```

### Steps to reproduce

```bash
1. add stream_chat_flutter: ^6.3.0 to your project
2. run flutter debug on iPhone emulator/android simulator
```

### Relevant log output

```shell
Failed to build iOS app
Error (Xcode): ../../../.pub-cache/hosted/pub.dev/stream_chat_flutter-6.3.0/lib/src/scroll_view/photo_gallery/stream_photo_gallery_tile.dart:207:5: Error: Type 'ImageDecoderCallback' not found.
stream_photo_gallery_tile.dart:207
Could not build the application for the simulator.
```

### Flutter doctor output

```shell
[✓] Flutter (Channel stable, 3.7.12, on macOS 13.3.1 22E772610a darwin-arm64, locale en-IN)
    • Flutter version 3.7.12 on channel stable at /Users/chinmay/fvm/versions/3.7.12
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (7 weeks ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/chinmay/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/chinmay/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.79.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 19C28C2C-BE82-4C70-B72F-CA66A56EF2BD • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 13.3.1 22E772610a darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 114.0.5735.106

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by xsahil03x** [comment]
Hey @chinmaygit, can you update your flutter to latest version? It should be fixed.

Feel free to re-open if it persists.

**Comment by chinmaygit** [comment]
Hi @xsahil03x Unfortunately we cannot update our flutter to 3.10, We have policy to keep flutter version to latest -1.

This makes stream_chat_flutter v6 implicitly tide with flutter 3.10. Is this expected behaviour? 

I suggest to still use deprecated `DecoderBufferCallback`, because `ImageDecoderCallback ` is new addition to the latest `painting/image_provider.dart`. This will cause error for the rest versions.
```
@override
  ImageStreamCompleter loadBuffer(
    MediaThumbnailProvider key,
    // ignore: deprecated_member_use
    DecoderBufferCallback decode,
  ) {
 ```

**Comment by xsahil03x** [comment]
Working on a solution.

**Comment by chinmaygit** [comment]
Thanks a ton @xsahil03x. When can we expect a new version?

**Comment by xsahil03x** [comment]
@chinmaygit Monday probably.

**Comment by chinmaygit** [comment]
Hello @xsahil03x, When can we expect the new version with this fix?

**Comment by xsahil03x** [comment]
@chinmaygit preparing it.

**Comment by rohan20** [comment]
Hey @xsahil03x, I understand that you must be busy with multiple things. Could you please let us know a tentative ETA so that we can plan our work accordingly? Thank you 😄

**Comment by xsahil03x** [comment]
@rohan20 Its already available as part of [v6.4.0](https://pub.dev/packages/stream_chat_flutter/changelog#640)
